### PR TITLE
Avoid ClassfileAnnotation warning for @SerialVersionUID

### PR DESCRIPTION
**Pull request by soc**
@SerialVersionUID is special-cased, the warning doesn't apply.

Related to SI-7041.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -1720,7 +1720,10 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       if ((clazz isNonBottomSubClass ClassfileAnnotationClass) && (clazz != ClassfileAnnotationClass)) {
         if (!clazz.owner.isPackageClass)
           context.error(clazz.pos, "inner classes cannot be classfile annotations")
-        else restrictionWarning(cdef.pos, unit,
+        // Ignore @SerialVersionUID, because it is special-cased and handled completely differently.
+        // It only extends ClassfileAnnotationClass instead of StaticAnnotation to get the enforcement
+        // of constant argument values "for free".
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Could you add a reference to SI-7041 in the comment?

**Comment by gkossakowski** [comment]
@soc, will you have a chance to fix the comment today? Would like to :ship: this as part of 2.11.3 release.

**Comment by xeno-by** [comment]
:ship: for the win :D

**Comment by soc** [comment]
@lrytz @gkossakowski @xeno-by Done.

**Comment by gkossakowski** [comment]
Oh wow! I didn't realize that there's a long history behind this change. Before I start digging I may ask @lrytz for opinion.

**Comment by lrytz** [comment]
LGTM! Thanks, @soc

**Pull request merged**